packages/transactions/src/clarity/prettyPrint.ts

```diff
@@ -64,7 +66,9 @@ function formatTuple(cv: TupleCV, space: number, depth = 1): string {
   const spaceBefore = formatSpace(space, depth, false);
   const endSpace = formatSpace(space, depth, true);
 
-  return `{${spaceBefore}${items.join(`,${spaceBefore}`)}${endSpace}}`;
+  return `{${spaceBefore}${items
+    .sort((a, b) => a[0].localeCompare(b[0]))
```

Comment:
Are you confident that `localCompare` will always perform the correct lexicographical ordering? I think when it's called without a locale like this, then it defaults to the environments locale (e.g. the user's browser locale settings)

Reply:
I used it because it enable case-insensitive string sorting. 
But indeed it might be more consistent if I enforce the locale to "en", if we consider that other different locales could result in different sorting?

Also, what really matters here in my opinion is that to identical input result in the same output
(eg: `{b: 2, a: 1}` and `{a: 1, b: 2}` should both give the same output)

Reply:
So I'm also happy with just `sort()`, developrs should mix upper/lower case in the tuples anyway

Reply:
Does core do case-insensitive sorting? I thought it was purely lexicographical. Either way, I think this is the typical way to do that in js:
```js
.sort((a, b) => a.toLowerCase() > b.toLowerCase() ? 1 : (a.toLowerCase() < b.toLowerCase() ? -1 : 0));
```

Reply:
How about `a.localeCompare(b, 'en')`? We can set the locale to be more deterministic. (with an optional ` { sensitivity: 'base' }` for case-insensitive.
But is the `[0]` correct here, or just comparing the first character?

Reply:
> Does core do case-insensitive sorting?

What do you mean by `core`?

If the snippet you shared and `localeCompare` return the same result, shouldn't we prefer `localeCompare`?

Replying to @janniks comment as well:
> maybe let's add a note to not use the function for keeping keys in a specific order

Or I can enforce the locale to `en` and the sorting will always be the same

Reply:
> What do you mean by core?

The clarity runtime(s)

> If the snippet you shared and localeCompare return the same result, shouldn't we prefer localeCompare?

It seems like localCompare could return different results depending on host environment. Also AFAIK there's not a guarantee that every environment will have every locale (maybe `en` is an exception?), I've seen slim nodejs environments with seemingly no locale installed.


The snippet I shared uses unicode scalar value comparisons which is essentially the same as lexicographical comparison in this case, I think. If you are confident about the behavior of `localCompare` then feel free to use it :)

Reply:
> Does core do case-insensitive sorting?

In clarity the order of the tuple keys are not sorted at all

>  If you are confident about the behavior of localCompare

Well I was but now I'm not (with " slim nodejs environments with seemingly no locale installed".

I'm tempted to go with just `items.sort()` now. If that was the initial implementation, would it have been fine for both of you?

Reply:
Keys are always strings in Clarity, right?

Reply:
@janniks yes, and better than that, it must start with `[a-zA-Z]`